crates/loans/src/lib.rs

```diff
@@ -950,22 +997,13 @@ pub mod pallet {
         #[transactional]
         pub fn deposit_all_collateral(origin: OriginFor<T>, asset_id: AssetIdOf<T>) -> DispatchResultWithPostInfo {
             let who = ensure_signed(origin)?;
-            let lend_tokens = Self::free_lend_tokens(asset_id, &who)?;
-            ensure!(!lend_tokens.is_zero(), Error::<T>::InvalidAmount);
-            Self::do_deposit_collateral(&who, lend_tokens.currency(), lend_tokens.amount())?;
-            Ok(().into())
-        }
-
-        #[pallet::weight(<T as Config>::WeightInfo::collateral_asset())]
-        #[transactional]
-        pub fn deposit_collateral(
-            origin: OriginFor<T>,
-            asset_id: AssetIdOf<T>,
-            #[pallet::compact] amount: BalanceOf<T>,
-        ) -> DispatchResultWithPostInfo {
-            let who = ensure_signed(origin)?;
-            ensure!(!amount.is_zero(), Error::<T>::InvalidAmount);
-            Self::do_deposit_collateral(&who, asset_id, amount)?;
+            let free_lend_tokens = Self::free_lend_tokens(asset_id, &who)?;
+            ensure!(!free_lend_tokens.is_zero(), Error::<T>::InvalidAmount);
```

Comment:
We should probably return a more descriptive error here - it'll be confusing for users since they did not specify any amount. Also, this error will also be thrown if they already called `deposit_all_collateral` previously, right?

Reply:
It's not the best worded error but it was already in use everywhere else in the pallet and its doc comment is `Amount cannot be zero`. Maybe I should just rename the error to `AmountIsZero`?

Reply:
But my point is that `Amount cannot be zero` is confusing since the user is not supplying any explicit amount as argument.

Ideally we'd have something like `AlreadyConfiguredAsCollateral` when they already have the toggle enabled, or `BalanceZero` (with docstring like "Operation unsupported when balance is zero.") when they just try to call this without actually having the token

Reply:
This is something the UI and lib are on the way of preventing as part of input validation, but I did rephrase the error messages.